src/lib.rs

```diff
@@ -187,6 +227,12 @@ where
         self.await_machine_state(target)
     }
 
+    /// Resets the chip.
+    pub fn reset(&mut self) -> Result<(), Error<SpiE, GpioE>> {
+        self.0.write_strobe(Command::SRES)?;
+        Ok(())
+    }
+
     /// Configure some default settings, to be removed in the future.
     #[rustfmt::skip]
     pub fn set_defaults(&mut self) -> Result<(), Error<SpiE, GpioE>> {
```

Comment:
just a small note, this set_defaults was just a temporary hack that should go away once the functions have been added for all of this.

Reply:
Sure. I kept it with the same behaviour for backwards compatibility, just factoring out the bits that now have methods. I figure removing it can be a separate PR once everything else is done, if you want.